src/resource_manager/secret.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""gateway-api-integrator secret resource manager."""
+
+
+import logging
+import typing
+
+from cryptography.hazmat.primitives import serialization
+from lightkube import Client
+from lightkube.core.client import LabelSelector
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.resources.core_v1 import Secret
+from lightkube.types import PatchType
+
+from state.base import State
+
+from .permission import map_k8s_auth_exception
+from .resource_manager import ResourceManager
+
+logger = logging.getLogger(__name__)
+
+
+def _get_decrypted_key(private_key: str, password: str) -> str:
+    """Decrypted the provided private key using the provided password.
+
+    Args:
+        private_key: The encrypted private key.
+        password: The password to decrypt the private key.
+
+    Returns:
+        The decrypted private key.
+    """
+    decrypted_key = serialization.load_pem_private_key(
+        private_key.encode(), password=password.encode()
+    )
+
+    # There are multiple representation PKCS8 is the default supported by nginx controller
+    return decrypted_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode()
+
+
+class SecretResourceManager(ResourceManager[Secret]):
+    """Kubernetes Ingress resource controller."""
+
+    def __init__(self, labels: LabelSelector, client: Client) -> None:
+        """Initialize the SecretResourceManager.
+
+        Args:
+            labels: Label to be added to created resources.
+            client: Initialized lightkube client.
+        """
+        self._client = client
+        self._labels = labels
+
+    @property
+    def _name(self) -> str:
```

Comment:
Would you be able to, in the docstrings, perhaps mention that this is to fulfill the ResourceManager protocol please? :D

Reply:
I have a PR coming that will completely remove this unused property so I'm resolving this comment for now.